lib/client.js

```diff
@@ -62,24 +51,29 @@ class Parser extends HTTPParser {
     const request = client[kQueue][client[kComplete]]
     const skipBody = request.method === 'HEAD'
 
-    const body = new Readable({
-      autoDestroy: true,
-      read: this.bodyRead,
-      destroy: this.bodyDestroy
-    })
-    body.push = request.wrapSimple(body, body.push)
-
-    this.body = body
     this.read = 0
-
-    request.callback(null, {
+    this.body = request.callback(null, {
       statusCode,
-      headers: parseHeaders(headers),
-      body
+      headers: parseHeaders(headers)
     })
 
-    if (skipBody) {
-      this[HTTPParser.kOnMessageComplete]()
+    if (this.body) {
+      if (skipBody) {
+        this.body.end()
+        this.body = null
+      } else {
+        this.body
+          .on('drain', this.resume)
+          // TODO: 'error' and 'close' should only resume if waiting for 'drain'.
+          .on('error', this.resume)
+          .on('close', this.resume)
+        this.body.write = request.wrapSimple(this.body, this.body.write)
+        this.body.end = request.wrapSimple(this.body, this.body.end)
```

Comment:
Both wraps are needed otherwise async_hooks tests fail

Reply:
perfect, let's go with the wraps.